### PR TITLE
Add Segment.io Telemetry

### PR DESCRIPTION
**Pull request by roborourke**
Porting Accelerate plugin telemetry system to core.

Related to https://github.com/humanmade/product-dev/issues/874

Screenshot 2022-01-14 at 16 30 44

### CODE REVIEW

**Comment by roborourke** [comment]
Actually we do have some UI stuff in this module already so I'll add the basic opt in UI here too.

**Comment by roborourke** [comment]
It's basic styling but we can remove it in the analytics plugin in favour of the React UI.

File: inc/telemetry/namespace.php
Comment (tfrommen) [inline inc/telemetry/namespace.php:194]: This might be a bit edge-case, but it's possible that two or more sites have the same registration timestamp, if done programmatically. Why not use something that is actually required to be unique, like the ID full domain? 🤔 (Maybe I don't understand how/what this is used for, or why the date is being used to generate the hash.)

Comment (rmccue) [reply inc/telemetry/namespace.php:194]: These are user IDs across development machines, so ID = 1 and domain is going to overlap across all devs on the same project generally.

File: inc/telemetry/namespace.php
Comment (tfrommen) [inline inc/telemetry/namespace.php:234]: Does this change (often)? Would it be better moved to an actual JavaScript file?

**Review by rmccue — APPROVED** [review]
Looks good to me but I feel like I'm partially just reviewing my own code so you might want a second review?

File: inc/telemetry/namespace.php
Comment (tfrommen) [reply inc/telemetry/namespace.php:194]: Aha, I see. You want to distinguish multiple local development environments...?

File: inc/telemetry/namespace.php
Comment (rmccue) [reply inc/telemetry/namespace.php:234]: This is Segment's snippet which we use verbatim from them.

Comment (roborourke) [reply inc/telemetry/namespace.php:234]: Well potentially we could punt it to tag manager instead but for the time being this is fine, changing it is pretty easy and backporting if needed is easy too.

**Pull request merged**
